### ISSUE TITLE
NPE during delete edge

### ISSUE THREAD

**Issue by mriehl** [orientechnologies/orientdb#5603]
I'm trying to clean up some broken edges where the vertex in direction `OUT` has `OrientVertex::getRecord() == null`.

My problem is that `OrientGraph::removeEdgeInternal` does not handle the NPE.
I get the following stack trace:

```
2016-01-18 15:38:10:977 WARNI Will remove broken edge e[#58:11319][#47:115423-LIA->#41:15725] [DPlugin$2]Failed to remove broken edge e[#58:11319][#47:115423-LIA->#41:15725]
java.lang.NullPointerException
        at com.tinkerpop.blueprints.impls.orient.OrientGraph.removeEdgeInternal(OrientGraph.java:402)
        at com.tinkerpop.blueprints.impls.orient.OrientEdge.remove(OrientEdge.java:380)
        at com.tinkerpop.blueprints.impls.orient.OrientBaseGraph.removeEdge(OrientBaseGraph.java:1155)
        at org.orientdb.DPlugin$2.checkEdge(DPlugin.java:197)
        at org.orientdb.DPlugin$2.execute(DPlugin.java:177)
        at com.orientechnologies.orient.core.sql.functions.OSQLFunctionRuntime.execute(OSQLFunctionRuntime.java:127)
        at com.orientechnologies.orient.core.sql.ORuntimeResult.applyRecord(ORuntimeResult.java:116)
        at com.orientechnologies.orient.core.sql.ORuntimeResult.getProjectionResult(ORuntimeResult.java:222)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.addResult(OCommandExecutorSQLSelect.java:607)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.handleNoTarget(OCommandExecutorSQLSelect.java:2461)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execute(OCommandExecutorSQLSelect.java:429)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:90)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1547)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:1528)
        at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:292)
        at com.orientechnologies.orient.core.sql.query.OSQLQuery.run(OSQLQuery.java:72)
        at com.orientechnologies.orient.core.sql.query.OSQLSynchQuery.run(OSQLSynchQuery.java:85)
        at com.orientechnologies.orient.core.query.OQueryAbstract.execute(OQueryAbstract.java:33)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1323)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:400)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:223)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
```

I'm using `OrientGraph::removeEdge` to remove the edges once I have found them.
Is there a workaround or another way to get rid of these pesky edges?
Thanks a lot for your help.

In `OrientGraph` the NPE happens here:

```java
outVertexRecord.field(outFieldName));
```

so I think that `outVertexRecord` is `null`.

**Comment by mriehl** [comment]
I think that 

```java
   if (outVertex != null) {
      if (outVertex != null) {
```
in `OrientGraph` is wrong, one of these checks should ensure that `outVertex.getRecord()` is not `null`.
So it looks like the `else` clause was designed to deal with this, but it's never used. The code would then proceed to drop the other stuff which should work (the vertex in direction `IN` exists for example).

I could put together a pull request if everyone agrees that's what should happen.

**Comment by mriehl** [comment]
Just noticed this is already fixed in the `develop` branch. Closing, sorry.

**Comment by mriehl** [comment]
Actually the fix is in 85ec41b942ccae8321099b3086732cb43853deca . But it has not been backported to 2.1.x yet.
@lvca what's the process for backporting? Do you want me to cherry pick it into the 2.1.x branch and provide a pull request?

**Comment by mriehl** [comment]
I cherry-picked the commit into the `2.1.9` branch and am using a release from there. It does solve my problem but it would be great to have the fix in official releases too ;-)

**Comment by luigidellaquila** [comment]
Hi @mriehl 

I think it can be safely backported and included in next releases, @lvca WDYT?

Thanks

Luigi

**Comment by lvca** [comment]
Absolutely +1. @luigidellaquila can you do that please?

**Comment by luigidellaquila** [comment]
Done! The fix will be released today with 2.1.10. 

Thanks

Luigi

**Comment by mriehl** [comment]
:+1: thank you both!
